tests/unit/schema/test_api.py

```diff
@@ -2602,3 +2607,374 @@ def test_has_iam_mixin():
+    """
+    Tests the `gapic.schema.api.all_method_settings` method which reads
+    `MethodSettings` from the service config YAML.
+    https://github.com/googleapis/googleapis/blob/7dab3de7ec79098bb367b6b2ac3815512a49dd56/google/api/client.proto#L325
+    """
+    service_yaml_config = {
+        "apis": [
+            {"name": "google.example.v1beta1.SomeExample.Example1"},
+        ],
+        "publishing": {
+            "method_settings": [
+                {
+                    "selector": "google.example.v1beta1.SomeExample.Example1",
+                    "auto_populated_fields": [
+                        "squid",
+                        "mollusc",
+                    ],
+                },
+            ]
+        },
+    }
+    cli_options = Options(service_yaml_config=service_yaml_config)
+    field_options = descriptor_pb2.FieldOptions()
+    field_options.Extensions[
+        field_info_pb2.field_info
+    ].format = field_info_pb2.FieldInfo.Format.Value("UUID4")
+
+    squid = make_field_pb2(
+        name="squid", type="TYPE_STRING", options=field_options, number=1
+    )
+    mollusc = make_field_pb2(
+        name="mollusc", type="TYPE_STRING", options=field_options, number=2
+    )
+    fields = [squid, mollusc]
+    fd = get_file_descriptor_proto_for_method_settings_tests(fields=fields)
+    api_schema = api.API.build(fd, "google.example.v1beta1", opts=cli_options)
+    assert api_schema.all_method_settings == {
+        "google.example.v1beta1.SomeExample.Example1": client_pb2.MethodSettings(
+            selector="google.example.v1beta1.SomeExample.Example1",
+            auto_populated_fields=["squid", "mollusc"],
+            long_running=client_pb2.MethodSettings.LongRunning(),
+        )
+    }
+
+
+def test_method_settings_duplicate_selector_raises_error():
+    """
+    Test that `MethodSettingsError` is raised when there are duplicate selectors in
+    `client_pb2.MethodSettings`.
+    """
+    fd = get_file_descriptor_proto_for_method_settings_tests()
+    api_schema = api.API.build(fd, "google.example.v1beta1")
+    methodsettings = [
+        client_pb2.MethodSettings(
+            selector="google.example.v1beta1.SomeExample.Example1",
+        ),
+        client_pb2.MethodSettings(
+            selector="google.example.v1beta1.SomeExample.Example1",
+        ),
+    ]
+    with pytest.raises(
+        api.MethodSettingsError, match="(?i)duplicate selector"
+    ):
+        api_schema.enforce_valid_method_settings(methodsettings)
+
+
+def test_method_settings_invalid_selector_raises_error():
+    """
+    Test that `MethodSettingsError` when `client_pb2.MethodSettings.selector`
+    cannot be mapped to a method in the API.
+    """
+    fd = get_file_descriptor_proto_for_method_settings_tests()
+    api_schema = api.API.build(fd, "google.example.v1beta1")
+    methodsettings = [
+        client_pb2.MethodSettings(
+            selector="google.example.v1beta1.DoesNotExist.Example1",
```

Comment:
nit: Do you want to test a non-existent method in a valid service as well?

Reply:
Fixed in https://github.com/googleapis/gapic-generator-python/pull/1975/commits/eac51b69af4a2f3b03102a3e609f62e442a52e3e